rust/gelato/src/sponsored_call.rs

```diff
@@ -0,0 +1,142 @@
+use crate::types::Chain;
+use crate::{types::serialize_as_decimal_str, RELAY_URL};
+use ethers::types::{Address, Bytes, U256};
+use serde::{Deserialize, Serialize};
+use tracing::instrument;
+
+#[derive(Debug, Clone, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct SponsoredCallArgs {
+    pub chain_id: Chain,
+    pub target: Address,
+    pub data: Bytes,
+
+    // U256 by default serializes as a 0x-prefixed hexadecimal string.
+    // Gelato's API expects the gasLimit to be a decimal string.
+    #[serde(serialize_with = "serialize_as_decimal_str")]
+    pub gas_limit: Option<U256>,
+    pub retries: Option<u32>,
+}
+
+#[derive(Debug, Clone)]
+pub struct SponsoredCallCall<'a> {
```

Comment:
Uhh, `...CallCall`?
Can we get some brief docs at least on the intent of these structs.

Reply:
lmao you're totally right. This is the result of a `s/ForwardRequest/SponsoredCall/g`. I'll rename to `SponsoredCallApiCall` which is still not perfect but at least slightly less horrible